File: service/src/main/java/net/huray/backend/minuting/service/user/UserService.java
package net.huray.backend.minuting.service.user;

import java.util.UUID;
import lombok.RequiredArgsConstructor;
import net.huray.backend.minuting.account.Member;
import net.huray.backend.minuting.service.space.SpaceComponent;
import net.huray.backend.minuting.web.home.data.BaseInfoRes;
import net.huray.backend.minuting.web.home.data.SpaceRes;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class UserService {

    private final UserComponent userComponent;
    private final SpaceComponent spaceComponent;

    public BaseInfoRes getBaseInfo(UUID uid) {
        Member member = userComponent.get(uid);
        return BaseInfoRes.of(member,
            spaceComponent.listPermissionByMember(member).stream()
            .map(e -> SpaceRes.from(e.getSpace()))
                .toList());
    }
}

File: service/src/main/kotlin/net/huray/backend/minuting/service/UserService.kt
package net.huray.backend.minuting.service

import net.huray.backend.minuting.dto.CompanyDto
import net.huray.backend.minuting.dto.SpaceDto
import net.huray.backend.minuting.dto.TeamDto
import net.huray.backend.minuting.dto.UserInfoDto
import net.huray.backend.minuting.service.component.SpaceComponent
import net.huray.backend.minuting.service.component.UserComponent
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserService(
    private val userComponent: UserComponent,
    private val spaceComponent: SpaceComponent
) {

    fun getBaseInfo(uid: UUID) = userComponent.get(uid)
        .run {
            UserInfoDto.UserInfoDetail(
                name, memberType,
                CompanyDto.SimpleRes(company.id, company.name, company.ceo, company.telNumber, company.address),
                TeamDto.SimpleRes(id = team.id, name = team.name),
                spaceComponent.listPermissionByMember(this).map { SpaceDto.SimpleRes(it.space) }
            )
        }

    fun listByName(name: String) = userComponent.searchByName(name)
        .map {
            UserInfoDto.UserInfoSimple(it.uid, it.name)
        }

}